File: recursion_2.py
#!/usr/bin/python3

def isArrayInSortedOrder(A):
    """
    Check if the array is in sorted order
    with Recursion.
    """
    if len(A) == 1:
        return True
    return A[0] <= A[1] and isArrayInSortedOrder(A[1:])


arr = [1, 2, 3, 4, 5, 6, 7, 8]
print(isArrayInSortedOrder(arr))

File: recursion_2.js
#!/usr/bin/node

isSorted = (arr) => {
    if (arr.length == 1)
    {
        return true;
    }
    return arr[0] <= arr[1] && isSorted(arr.slice(1, arr.length))
} 

arr = [1, 2, 3, 4, 5, 6, 7, 8]
console.log(isSorted(arr))

// arr = ["Apple", "Banana", "Cherry", "Pear", "Orange"]
// console.log(arr)

// slc = arr.slice(1, arr.length)
// console.log(slc)

// console.log(arr.length)